hazelcast/src/main/java/com/hazelcast/jmx/LocalStatsDelegate.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jmx;
+
+import com.hazelcast.jmx.suppliers.StatSupplier;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class LocalStatsDelegate<T> {
+
+    private volatile T localStats;
+    private final StatSupplier<T> supplier;
+    private final long intervalMs;
+    private final AtomicLong lastUpdated = new AtomicLong(0);
+    private final AtomicBoolean inProgress = new AtomicBoolean(false);
+
+    public LocalStatsDelegate(StatSupplier<T> supplier, long intervalSec) {
+        this.supplier = supplier;
+        this.intervalMs = intervalSec * 1000;
+    }
+
+    public T getLocalStats() {
+        long delta = System.currentTimeMillis() - lastUpdated.get();
+        if (delta > intervalMs) {
+            if (inProgress.compareAndSet(false, true)) {
+                localStats = supplier.get();
```

Comment:
what if we get an exception here?
In an exceptional case, the below part will not be executed.
```
lastUpdated.set(System.currentTimeMillis());
inProgress.set(false);
```